### DISCUSSION TITLE
Login: Remember me for always

### DISCUSSION THREAD

**Discussion by superfunk2000** [Lissy93/dashy#236] [Ideas]
I miss the "always" option when I login. I would be happy if that were possible.

**Comment by lissy93** [comment]
Heya,
So it's not actually possible to create a cookie with no expiration date. If the expiry date isn't specified, it will clear when the session ends. If you set a timestamp for ages in the future, your browser usually ends up deleting the cookie after a while, and you'll be logged out.

 But I've made you a PR #237 which adds a new option to the dropdown, that will last for a long time. So pull from `FEATURE/login-remember-me-always`, and see how that works out for you. Hope that helps :) 

Also, I recommend that you take a look at the [Keycloak auth method](https://github.com/Lissy93/dashy/blob/master/docs/authentication.md#keycloak), it is more secure, and has way more features than I could ever implement into the basic auth- I think you'll like it :) Or you could use your own SSO system, like [Authelia](https://github.com/Lissy93/dashy/blob/master/docs/authentication.md#authelia).

**Comment by superfunk2000** [comment]
Please excuse me for having to ask so stupidly, what is the name of the image that I have to enter in docker-compose?

**Reply by lissy93** [reply]
Oh, sorry thought you were building from source. I've created you a tag on DH, try this: `feature-237`.
So you should be able to use this with `docker pull lissy93/dashy:feature-237`, and then proceed as normal to spin up the container.

**Comment by superfunk2000** [comment]
That's it, thank you!

**Reply by lissy93** [reply]
This is now merged into the main release in #237 / V-1.9.3 :)

**Comment by wychwitch** [comment]
Hello! I hope this isn't considered necroposting, but I was running into this same issue, found this solution but when running the docker pull command I got this error:

`Error response from daemon: manifest for lissy93/dashy:feature-237 not found: manifest unknown: manifest unknown`

Is this image no longer supported? If not, could a new one be made? Just having the option for something longer than a week would help a lot!

**Reply by lissy93** [reply]
Correct, this image isn't supported any more. If this is a feature you will find useful, then I can include it in the main release? But keep in mind that since it's not actually possible to store a cookie 'for ever', so at some point, you will be logged out.

**Reply by wychwitch** [reply]
> Correct, this image isn't supported any more. If this is a feature you will find useful, then I can include it in the main release? 

Yes please!! Even if it's just a config setting to set it that way I would be happy 

> But keep in mind that since it's not actually possible to store a cookie 'for ever', so at some point, you will be logged out.

Ah don't worry I understand!! It's fine if I have to relog in every once in a while, it's just that *every week* was getting a little annoying hahaha. Even setting it for 2 weeks would be enough for me, but the longer the better for my desktop 

Thank you so much!!

**Reply by lissy93** [reply]
Cheers @WychWitch - This is now merged in #237 :)
Once V 1.9.3 is built, then you should see the update in your app.

**Reply by wychwitch** [reply]
> Cheers @WychWitch - This is now merged in #237 :)
> Once V 1.9.3 is built, then you should see the update in your app.

Thank you so much!!! I'll update as soon as I can!!

**Reply by wychwitch** [reply]
Sorry for taking so long to updated, finals hit me hard hahaha

I updated by stopping and removing the previous docker container and pulling the new image with `sudo docker pull lissy93/dashy:latest`, and after relaunching with the following command, it still says its on 1.9.2. I'm not sure what's wrong? I even tried with release-1.9.3 but it didn't work either. The new login option wasn't there either

```
docker run -d \
  -p 7080:80 \
    --name my-dashboard \
  --restart=unless-stopped \
  lissy93/dashy:latest
  ```
  
Any advice you can give me I'd appreciate @Lissy93 !! I made sure to hard refresh as well

**Reply by lissy93** [reply]
Heya, so it sounds like you've not got the latest version, but if you run `docker ps -a`, what version do you see?

Usually to manually update, pulling the latest image, then stopping, removing and recreating the container should work. If for some reason it isn't working, try adding the `--no-cache` flag to the pull command.

For more detailed instructions, try the steps in the [Updating Docs](https://github.com/Lissy93/dashy/blob/master/docs/management.md#updating) and this should work. But I think you'll fine [Watchtower](https://github.com/containrrr/watchtower) useful, as it will keep all your running containers up-to-date automatically, and is much easier in the long run.

Anyway, let me know if that still doesn't work, and I'll try and help further. And good luck on the finals! :)

P.s, if you're finding the project useful, Starring Dashy on GitHub will help new users find it :)

**Reply by wychwitch** [reply]
> 

OH WEIRD I went to check and it was 1.9.3, and when I checked the site again it indeed was correct on the site as well!! I must've just ran into some weird caching issue, thank you for the tips tho!! And thanks for watchtower, I have a lot of containers so that sounds so much easier ^^
